packages/agent-toolkit/src/utils/tools/initializing.utils.ts

```diff
@@ -0,0 +1,13 @@
+import { ApiClient } from '@mondaydotcomorg/api';
+import { Tool } from 'src/core/tool';
+import { BaseMondayApiTool } from 'src/core/tools/platform-api-tools/base-monday-api-tool';
+
+export const createToolInstance = (
+  tool: new (...args: any[]) => Tool<any, any>,
+  instanceOptions: { apiClient: ApiClient; apiToken: string },
+) => {
+  if (tool.prototype instanceof BaseMondayApiTool) {
+    return new tool(instanceOptions.apiClient);
+  }
+  return new tool();
```

Comment:
but the mondayAppsTool is getting apiToken, no?

Reply:
Yeah, did that in the second PR as it doesn't exist in this PR